src/query/functions/src/scalars/boolean.rs

```diff
@@ -338,7 +341,109 @@ pub fn register(registry: &mut FunctionRegistry) {
                     );
             }
             _ => unreachable!(),
-        });
+        })
+    }
+
+    for src_type in ALL_FLOAT_TYPES {
+        with_float_mapped_type!(|NUM_TYPE| match src_type {
+            NumberDataType::NUM_TYPE => {
+                registry.register_1_arg::<NumberType<NUM_TYPE>, BooleanType, _, _>(
+                    "to_boolean",
+                    |_, domain| {
+                        FunctionDomain::Domain(BooleanDomain {
+                            has_false: domain.min <= OrderedFloat(0.0)
+                                && domain.max >= OrderedFloat(0.0),
```

Comment:
Is there any difference from:
```suggestion
                            has_false: domain.min == OrderedFloat(0.0)
                                && domain.max == OrderedFloat(0.0),
```